cloud-controller-manager/do/loadbalancers.go

```diff
@@ -411,6 +451,54 @@ func (l *loadBalancers) EnsureLoadBalancerDeleted(ctx context.Context, clusterNa
 		return fmt.Errorf("failed to delete load-balancer: %s", err)
 	}
 
+	domain, err := getDomain(service)
+	if err != nil {
+		klog.Errorf("failed to fetch domain from service for cleanup: %s", err)
+		return nil
+	}
+
+	// attempt to clean up any managed A records or certificates related to LB
+	if domain != nil && lb != nil {
+		certID := getCertificateIDFromLB(lb)
+		certName := getCertificateName(domain.full)
+		cert, resp, err := l.resources.gclient.Certificates.Get(ctx, certID)
+		if err != nil && resp.StatusCode != http.StatusNotFound {
+			klog.Errorf("failed to fetch certificate: %s", err)
```

Comment:
Should this be a `return`?

Reply:
Let's include the certificate ID in the error message (`"failed to fetch certificate by ID %s: %s", certID, err`).

Similar below.

Reply:
I need to take a look upstream as to what happens when you return an error response in this method. It felt like the process shouldn't as long as the loadbalancer was successfully deleted. I will have to think more about whether we want strong guarantees around cleanup of these resources.